gensim/models/word2vec.py

```diff
@@ -1160,15 +1163,37 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
             if sentence_no % progress_per == 0:
                 logger.info(
                     "PROGRESS: at sentence #%i, processed %i words, keeping %i word types",
-                    sentence_no, total_words, len(vocab)
+                    sentence_no, total_words, max(len(vocab), word_counts.total())
                 )
-            for word in sentence:
-                vocab[word] += 1
+
             total_words += len(sentence)
 
-            if self.max_vocab_size and len(vocab) > self.max_vocab_size:
-                utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
-                min_reduce += 1
+            if use_bounter:
```

Comment:
bounter should be "drop-in" replacement (i.e. `if/else` looks suspicious here)

Reply:
does that mean the user should not be given a choice about whether bounter should be used.
(Especially considering bounter gives an approximate count)

Reply:
No, the user must choose what's he needed explicitly.